src/InheritDoc/package/build/SauceControl.InheritDoc.targets

```diff
@@ -3,24 +3,25 @@
 
 	<PropertyGroup>
 		<InheritDocEnabled Condition="'$(InheritDocEnabled)'=='' and '$(Configuration)'!='debug' and '$(DesignTimeBuild)'!='true' and '$(BuildingForLiveUnitTesting)'!='true'">true</InheritDocEnabled>
+		<CopyDocumentationFileToOutputDirectory Condition=" '$(InheritDocEnabled)' == 'true' ">false</CopyDocumentationFileToOutputDirectory>
 	</PropertyGroup>
 
-	<ItemGroup>
-		<_InheritDocBackupFile Include="@(DocFileItem -> '%(RootDir)%(Directory)%(Filename).original%(Extension)')" />
+	<ItemGroup Condition=" '$(InheritDocEnabled)' == 'true' ">
+		<!-- Force intermediate documentation file -->
+		<DocFileItem Remove="$(DocumentationFile)"/>
+		<DocFileItem Include="$(IntermediateOutputPath)$(AssemblyName).xml"/>
```

Comment:
This is not a good change.  The doc path may be set explicitly in the project file and may not be in the `obj` folder.

It is not guaranteed that `DocFileItem` and `FinalDocFile` are different paths, which is why `_InheritDocBackupFile` creates a unique name.

Reply:
We need to consider two kind of projects here:

1. Old projects explicitly specify `DocumentationFile`, typically in the `bin` folder.
In this case, `DocFileItem` and `FinalDocFile` coincide. Since `FinalDocFile` points to the `bin` folder due to its definition, we can guarantee different paths when we let `DocFileItem` point to the intermediate output path.
2. New projects specify `GenerateDocumentationFile` as `true` and the SDK lets `DocFileItem` point to the `obj` folder and `FinalDocFile` to the `bin` folder.
In this case, the redefinition of `DocFileItem` is a no-op.

I'm aware that this is a bit trickier than the `_InheritDocBackupFile` approach and might break customized builds more easily. For example, a customized build might have several items in `DocFileItem`.

I'd like to add some comments here.

Reply:
Ok, yeah, that makes sense.  I think this should be conditional on `DocFileItem` being the same as `FinalDocFile` so that it only catches projects of the first type and doesn't unintentionally change anything in projects of the second type.

So far, I only see it set here
https://github.com/dotnet/msbuild/blob/23b938b931b48da647e9dbca42323ee60abf1126/src/Tasks/Microsoft.Common.CurrentVersion.targets#L400